finger-exercises/react/ecmascript/02 - promises/index.js

```diff
@@ -0,0 +1,27 @@
+// Hint: use setInterval, create a new Promise and measure time with Date.now()
+
+export function delay(time) {
+  const start = Date.now();
+  return new Promise((resolve, reject) => {
+    if (time > 1000) {
+      reject(new Error('This time is too much !'));
+    }
+    const delayedTime = function delayed() {
+      resolve(Date.now() - start);
+    };
+    setInterval(delayedTime, time);
```

Comment:
Use `setTimeout` here. Otherwise, this function will be called every `time` milliseconds